lib/core/src/client/preview/loadCsf.ts

```diff
@@ -161,31 +161,41 @@ const loadStories = (
       if (isExportStory(key, meta)) {
         const storyFn = exports[key];
         const { story } = storyFn;
+        const { storyName = story?.name, parameters, decorators, args, argTypes } = storyFn;
+
+        const decoratorParams = decorators ? { decorators } : {};
+        const argsParams = { args, argTypes };
+
+        // storyFn.x and storyFn.story.x get merged with
+        // storyFn.x taking precedence in the merge
+        let deprecatedStoryParams = null;
         if (story) {
           logger.debug('deprecated story', story);
           deprecatedStoryAnnotationWarning();
+
+          deprecatedStoryParams = story.parameters;
+          if (story.decorators) {
+            decoratorParams.decorators = decoratorParams.decorators
+              ? [...decoratorParams.decorators, ...story.decorators]
+              : story.decorators;
+          }
+          if (story.args) {
+            argsParams.args = { ...story.args, ...argsParams.args };
+          }
+          if (story.argTypes) {
+            argsParams.argTypes = { ...story.argTypes, ...argsParams.argTypes };
+          }
         }
```

Comment:
How about just:

```js
const parameters = { ...storyFn.parameters, ...story.parameters };
const decorators = [...(storyFn.decorators || []), ...(story.decorators || [])];
// etc
```

The current code seems sort of convoluted.

Reply:
ok good suggestion. new code adds `decorators: [], args: {}, argTypes: {}` instead of `undefined` if none of those are provided. are you ok with that?

Reply:
@tmeasday self-merging for now. we can always make it smarter